MokManager.c

```diff
@@ -406,7 +409,32 @@ static void show_x509_info (X509 *X509Cert, UINT8 *hash)
 		return;
 
 	i = 0;
-	text = AllocateZeroPool(sizeof(CHAR16 *) * (fields*3 + 1));
+
+	extusage = X509_get_ext_d2i(X509Cert, NID_ext_key_usage, NULL, NULL);
+	text = AllocateZeroPool(sizeof(CHAR16 *) * (fields*3 + sk_ASN1_OBJECT_num(extusage)*3 + 1));
```

Comment:
This seems to be allocating more memory than what is needed (not necessarily a bad thing).

For the new strings related to extended key usage, I think it should only be an additional `sk_ASN1_OBJECT_num(extusage) + 2` that is *only* added when sk_ASN1_OBJECT_num(extusage) is greater than zero.